lsp/Main.hs

```diff
@@ -80,20 +80,14 @@ publishError uri err = publishDiagnostics 100 (toNormalizedUri uri) Nothing (par
 clearError :: MonadLsp config m => Uri -> m ()
 clearError uri = publishDiagnostics 100 (toNormalizedUri uri) Nothing (Map.singleton(Just "lexer") (Data.SortedList.toSortedList []))
 
-runDiagnosticsOnHover :: MonadLsp config m => Uri -> ExceptT Err IO a -> m ()
-runDiagnosticsOnHover uri ioHover = do
-  eitherHover <- liftIO $ runExceptT ioHover
-  case eitherHover of
+runDiagnosticsOnHover' :: MonadLsp config m => Uri -> Either Err b -> m ()
+runDiagnosticsOnHover' uri ioHover = do
```

Comment:
The names here are not very accurate anymore. Perhaps `sendDiagnosticsOnLeft` or something. And `ioHover` to `eitherHover`.